crates/bindings-sys/src/lib.rs

```diff
@@ -135,6 +168,10 @@ pub mod raw {
         ///
         /// The iterator is registered in the host environment
         /// under an assigned index which is written to the `out` pointer provided.
+        ///
+        /// Returns an error if
+        /// - a table with the provided `table_id` doesn't exist
+        /// - writing to `out` overflows an 64-bit integer
```

Comment:
For all the `_iter` variants, I thought `out` was a pointer-ish thing shared across the boundary, not an integer?